sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -71,18 +133,63 @@ enum {
  * - bite after escalation reset, so we should not get timer reset.
  */
 enum {
-  kWdogBarkMicros = 600,          // 600 us
-  kWdogBiteMicros = 800,          // 800 us
-  kEscalationStartMicros = 200,   // 200 us
-  kEscalationPhase0Micros = 200,  // 200 us
-  kEscalationPhase1Micros = 200,  // 200 us
-  kMaxResets = 2,
+  // Note that the escalation phase times below define the length of each phase,
+  // not when they start.
+  // The starting time is given by the aggregate of previous phase lengths, and
+  // is noted with @ below.
+  // @0 us -> in this phase we will not do anything so that the exception
+  // handlers have time to execute.
+  kEscalationPhase0Micros = 200,
+  // @200 us -> in this phase we will raise an NMI
+  kEscalationPhase1Micros = 200,
+  // @400 us -> in this phase we will assert lc_escalate_en
+  kEscalationPhase2Micros = 200,
+  // @600 us -> in this phase we will reset the chip
+  kEscalationPhase3Micros = 200,
+  // These are set so that both events happen in Phase2 of the escalation
+  // protocol, which asserts lc_escalate_en. That should prevent the Wdog
+  // from running and sending out an NMI on its own (we check in the NMI
+  // handler below that this does not happen).
+  kWdogBarkMicros = 450,
+  kWdogBiteMicros = 500,
+  // We expect exactly one reset
+  kMaxResets = 1,
   kMaxInterrupts = 30,
 };
 
-static_assert(kWdogBarkMicros < kWdogBiteMicros &&
-                  kWdogBarkMicros > kEscalationPhase0Micros,
-              "The wdog bite shall happen only if escalation reset fails.");
+static_assert(
+    kWdogBarkMicros < kWdogBiteMicros &&
+        kWdogBarkMicros > (kEscalationPhase0Micros + kEscalationPhase1Micros),
+    "The wdog bite shall after the NMI phase when lc_escalate_en is asserted.");
+/**
+ * Main SRAM addresses used in the sram_function_test.
+ */
+enum {
+  kSramStart = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR,
+  kSramEnd = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR +
+             TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_SIZE_BYTES,
+  kSramRetStart = TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR,
+};
+
+/**
+ * Program in main SRAM to jump to for testing corruption of instruction
+ * integrity.
+ *
+ * Using this program is convenient, since the sram_program.ld script ensures
+ * that the entry point will be mapped to the start of the main SRAM.
+ */
+OT_SECTION(".data.sram_function_test")
+OT_NOINLINE
+static void sram_function_test(void) {
+  uint32_t pc = 0;
+  asm("auipc %[pc], 0;" : [pc] "=r"(pc));
```

Comment:
Aha... I see you check the address is within SRAM in C :)

Reply:
Yeah, although this is only called in case the function is not corrupted (unless I move the corruption point past this check, which would actually be fun to try. let me see....